src/app/habits/createhabit.tsx

```diff
@@ -0,0 +1,238 @@
+              alert("create/edit habit");
+            } else {
+              setFailedRequirements(true);
+              alert(
+                `can't create habit:\n${needsTag ? " - missing tag\n" : ""}${needsName ? " - missing name" : ""}`,
+              );
+            }
+          }}
+        >
+          <Icon icon={habitid ? IconCheck : IconArrowForwardUp} size={20} />
+          <Text className="ml-1 text-base font-semibold">
+            {habitid ? "Done" : "Next"}
+          </Text>
+        </TouchableOpacity>
+      </View>
+
+      {/* Icon & Name */}
+      <View className="flex flex-col gap-y-2">
+        <Text className="text-base font-semibold">Icon & Name</Text>
+        <View className="flex flex-row">
+          <IconButton
+            icon={icons[icon]}
+            onPress={() => {
+              router.push("/habits/habiticons");
+            }}
+            required={needsTag && failedRequirements}
+          />
+          <TextInput
+            className="ml-2 h-10 flex-1 rounded-lg border px-3"
+            numberOfLines={1}
+            style={{
+              borderColor,
+              color: DefaultColors[colorScheme].text,
+              backgroundColor:
+                needsName && failedRequirements
+                  ? colors.habitColors.red.base
+                  : colors.transparent,
+            }}
+            placeholder="Ex. Workout"
+            placeholderTextColor={DefaultColors[colorScheme].placeholder}
+            value={habitName}
+            onChangeText={(text) => setHabitName(text)}
+          ></TextInput>
+        </View>
+      </View>
+
+      {/* Description */}
+      <View className="flex flex-col gap-y-2">
+        <Text className="text-base font-semibold">Description (optional)</Text>
+        <TextInput
+          className="h-10 rounded-lg border px-3"
+          style={{
+            borderColor,
+            color: DefaultColors[colorScheme].text,
+          }}
+          maxLength={50}
+          value={description}
+          onChangeText={(text) => setDescription(text)}
+        ></TextInput>
+      </View>
+
+      {/* Colour */}
+      <View className="flex flex-col gap-y-1">
+        <Text className="text-base font-semibold">Color</Text>
+        <View className="flex flex-row flex-wrap justify-between gap-x-1 gap-y-1">
+          {colorNames.map((col, idx) => (
+            <TouchableOpacity
+              className="flex h-9 w-9 items-center justify-center rounded-full"
+              key={idx}
+              style={{
+                backgroundColor:
+                  col === "stone"
+                    ? colors.stone[400]
+                    : colors.habitColors[col]?.base,
+              }}
+              onPress={() => {
+                setColor(col);
+              }}
+            >
+              {col === color ? <Icon icon={IconCheck} /> : null}
+            </TouchableOpacity>
+          ))}
+        </View>
+      </View>
```

Comment:
The format breaks on my phone (there are 3 rows of colors, and only 2 colors on the 3rd row), I think my phone is a bit smaller. Can you make the size of the circles automatic so that it works for any width?

An idea is to make two different flex rows, and in each you can use `flex-1` on the circles so that they resize to fill the container